lib/livebook_web/controllers/session_controller.ex

```diff
@@ -19,28 +19,42 @@ defmodule LivebookWeb.SessionController do
       {:ok, session} ->
         notebook = Session.get_notebook(session.pid)
 
-        send_notebook_source(conn, notebook, format)
+        file_name =
+          case session.file do
+            nil ->
+              notebook.name
+              |> String.downcase()
+              |> String.replace(" ", "_")
+
+            _ ->
+              %{"file_name" => file_name} =
+                Regex.named_captures(~r{.*[/](?<file_name>.*)[\.].*}, session.file.path)
+
+              file_name
+          end
```

Comment:
Suggestion: let's have a function called `Session.infer_file_name` or similar, so we can move this to a separate module and unit test it.

Reply:
Maybe something more specific like `file_name_for_download`, since we deal with several files already?

Sidenote, the `session` struct has `session.notebook_name`, so the infer function can take all information from there :)